src/compose/images.ts

```diff
@@ -108,6 +124,13 @@ export class Images extends (EventEmitter as {
 			return null;
 		} catch (e) {
 			if (!NotFoundError(e)) {
+				if (!(e instanceof ImageDownloadBackoffError)) {
+					this.imageFetchLastFailureTime[image.name] = Date.now();
```

Comment:
It might be better to use `process.hrtime()` here [like we did in the healthcheck functions](https://github.com/balena-io/balena-supervisor/blob/2c30b55a5c251f16ad1ffeb2b63c21091350f597/src/api-binder.ts#L120), to make this resilient to system clock changes (e.g. if ntp syncing is happening while we're downloading)